receiver/tlscheckreceiver/scraper.go

```diff
@@ -5,27 +5,85 @@ package tlscheckreceiver // import "github.com/open-telemetry/opentelemetry-coll
 
 import (
 	"context"
+	"crypto/tls"
+	"errors"
+	"sync"
+	"time"
 
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/receiver"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/tlscheckreceiver/internal/metadata"
 )
 
+var (
+	ErrMissingTargets = errors.New(`No targets specified`)
+)
+
 type scraper struct {
-	// include string
-	logger *zap.Logger
-	mb     *metadata.MetricsBuilder
+	cfg                *Config
+	settings           component.TelemetrySettings
+	mb                 *metadata.MetricsBuilder
+	getConnectionState func(host string) (tls.ConnectionState, error)
+}
+
+func getConnectionState(host string) (tls.ConnectionState, error) {
+	conn, err := tls.Dial("tcp", host, &tls.Config{InsecureSkipVerify: true})
+	if err != nil {
+		return tls.ConnectionState{}, err
+	}
+	defer conn.Close()
+	return conn.ConnectionState(), nil
 }
 
 func (s *scraper) scrape(_ context.Context) (pmetric.Metrics, error) {
-	return pmetric.NewMetrics(), nil
+	if s.cfg == nil || len(s.cfg.Targets) == 0 {
+		return pmetric.NewMetrics(), ErrMissingTargets
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(s.cfg.Targets))
+	var mux sync.Mutex
+	for _, target := range s.cfg.Targets {
+		go func(host string) {
+			defer wg.Done()
+
+			now := pcommon.NewTimestampFromTime(time.Now())
```

Comment:
Why calculated here and used long ago?

Reply:
Moved to when it is used to set collection timestamp